### PR TITLE
Run3-TB51 Correct the beam momentum generator for HGCal TB studies

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Correct the beam generator used for HGCal TB studies and update the analyzer to take the beam dependent parameters 

#### PR validation:

Use the cfg files in SimG4CMS/HGCalTestBeam/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: IOMC/ParticleGuns/src/BeamMomentumGunProducer.cc [hunk 1]
```diff
@@ -13,6 +12,8 @@
 #include "CLHEP/Random/RandFlat.h"
 
 #include "TFile.h"
+#include <cmath>
+#include <ostream>
```
Comment (kpedro88) [inline IOMC/ParticleGuns/src/BeamMomentumGunProducer.cc]: this include should not be needed, please remove it

File: IOMC/ParticleGuns/src/BeamMomentumGunProducer.cc [hunk 2]
```diff
@@ -104,23 +105,42 @@ namespace edm {
...
+      double pphi = pGeV.phi();
       double theta = CLHEP::RandFlat::shoot(engine, fMinTheta, fMaxTheta);
       double phi = CLHEP::RandFlat::shoot(engine, fMinPhi, fMaxPhi);
-      double px = mom * sin(theta) * cos(phi);
-      double py = mom * sin(theta) * sin(phi);
-      double pz = mom * cos(theta);
-
-      if (fVerbosity > 0)
-        edm::LogVerbatim("BeamMomentumGun") << "px:py:pz " << px << ":" << py << ":" << pz;
+      if (phi > M_PI)
```
Comment (kpedro88) [inline IOMC/ParticleGuns/src/BeamMomentumGunProducer.cc]: there are functions in https://github.com/cms-sw/cmssw/blob/master/DataFormats/Math/interface/deltaPhi.h to do this automatically, better to use them

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Comment by agrohsje** [comment]
@kpedro88 is your proposal to use the existing functions strict or not, i.e. is it just a proposal? Waiting for this before signing.

**Comment by kpedro88** [comment]
@agrohsje the PR was updated to use the deltaPhi functions where appropriate (some of the cases are slightly outside the scope of what is available centrally)

**Comment by agrohsje** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**